PlastomeRegionBurstAndAlign.py

```diff
@@ -340,165 +322,154 @@ def main(args):
-                # import subprocess
-                # subprocess.call(['mafft', '--auto', out_fn_unalign_nucl, '>', out_fn_aligned_nucl])
-
-                ## TO DO ##
-                # Automatically determine number of threads available #
-                # Have the number of threads saved as num_threads
-                num_threads = 1
-
-                ## TO DO ##
-                # Let user choose if alignment conducted with MAFFT, MUSCLE, CLUSTAL, etc.; use a new argparse argument and if statements in the ine below
-
-                alignm_cmdlexec = Bio.Align.Applications.MafftCommandline(input=out_fn_unalign_nucl,
-                                                                          adjustdirection=True, thread=num_threads)
-                stdout, stderr = alignm_cmdlexec()
-                with open(out_fn_aligned_nucl, 'w') as hndl:
-                    hndl.write(stdout)
-        else:
-            log.critical("No items in nucleotide main dictionary to process")
-            raise Exception()
-    # --------------------------------------------------------------------------#
-    # CONDUCT PROTEIN ALIGNMENT AND BACK-TRANSLATION TO NUCLEOTIDES
-    if select_mode == 'cds':
-        action = "conducting multiple sequence alignment based on protein sequence data, followed by a back-translation to nucleotides"
-        log.info("%s" % action)
-        ###
-        try:
-            proteinalign_and_backtranslate(main_d_prot, out_dir, log)
-        except:
-            raise Exception()
-    # --------------------------------------------------------------------------#
-    # CONVERT FASTA ALIGNMENT TO NEXUS ALIGNMENT AND APPEND TO LIST
+def conduct_protein_alignment_and_back_translation(main_d_prot, out_dir, log):
+    action = "conducting multiple sequence alignment based on protein sequence data, followed by back-translation to nucleotides"
+    log.info("%s" % action)
+    try:
+        proteinalign_and_backtranslate(main_d_prot, out_dir, log)
+    except:
+        raise Exception()
+
+def convert_fasta_alignment_to_nexus(main_d_nucl, out_dir):
     action = "converting FASTA alignment to NEXUS alignment"
     log.info("%s" % action)
-    ###
     successList = []
+
     for k in main_d_nucl.keys():
         aligned_nucl_fasta = os.path.join(out_dir, 'nucl_' + k + '.aligned.fasta')
         aligned_nucl_nexus = os.path.join(out_dir, 'nucl_' + k + '.aligned.nexus')
-        # Convert FASTA alignment to NEXUS alignment
         try:
-            Bio.AlignIO.convert(aligned_nucl_fasta, 'fasta', aligned_nucl_nexus, 'nexus', molecule_type='DNA')
+            AlignIO.convert(aligned_nucl_fasta, 'fasta', aligned_nucl_nexus, 'nexus', molecule_type='DNA')
         except:
             log.warning("Unable to convert alignment of `%s` from FASTA to NEXUS" % k)
-            # raise Exception()
-            continue  # skip to next k in loop, so that k is not included in successList
```

Comment:
Please keep comments such as this in the code while we develop it.

    # skip to next k in loop, so that k is not included in successList